### DISCUSSION TITLE
Host own web instance - gitpod is going away

### DISCUSSION THREAD

**Discussion by GitMensch** [eclipse-theia/theia#9301] [General]
... from Theia as vscode web is now stable enough :-(

As far as I've understood https://github.com/gitpod-io/gitpod/issues/3475 this decision is already "fix" and will be applied in the next months.

**Comment by vince-fugnitto** [comment]
@GitMensch thank you for the discussion, did you have a question?

**Comment by GitMensch** [comment]
No question, just the general discussion, bringing up that a self-hosted web-based Theia seems reasonable, given the current content of https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#run-the-browser-example-with-gitpod and the "Try" link at https://theia-ide.org - which would be both be wrong as soon as there's no Theia in Gitpod.

**Reply by vince-fugnitto** [reply]
@GitMensch the links you provided still work, they start the `example-browser` application in gitpod to test (only the gitpod editor is vscode going forward).

The workflow remains the same:
- prebuilt workspaces still exist.
- the links open gitpod (using vscode as an editor) - edit, browse code, debug, pull-requests.
- gitpod opens the `example-browser` instance of theia to test with (port 3000 for instance).

**Reply by GitMensch** [reply]
I possibly understood the "try" links wrong, using GitPod itself used to be the "try", but using this link, being in a vscode editor, then running the sample and get Theia on the right side next to it, is still working (I just never actually did that before...).
Maybe make that a bit more obvious, especially when clicking on the "Try" button on the main webpage that process was not clear to me [maybe just link from that page to the Developing.md link instead].

**Reply by vince-fugnitto** [reply]
> I possibly understood the "try" links wrong, using GitPod itself used to be the "try"...

That's incorrect. Gitpod allows us to automatically and easily checkout the project, build, and test the changes by starting the `example-browser` application. Gitpod itself was never the "try".

You can see it is described as the last step (we start the example-browser application):

![image](https://user-images.githubusercontent.com/40359487/113752468-0329b000-96db-11eb-82d3-9646d446bbe8.png)

**Reply by GitMensch** [reply]
I see. Maybe there's a way to show this when opening https://gitpod.io/#https://github.com/eclipse-theia/theia? Currently I see a huge README but not that screen.
In any case, thanks for pointing out "how to try correctly", it was possibly always totally clear and for some strange reason I got that wrong in the first place (and once you think that was the case....)

**Reply by vince-fugnitto** [reply]
I'm not sure what we should show?
The link ultimately:
- opens `gitpod` with `theia` as a workspace.
- opens a terminal where `theia` is built, and the `example-browser` will be started when ready.
- when completed it will prompt to open `theia` in a separate window (_open browser_).

I'm not sure where the confusion lies.

<div align='center'>

![image](https://user-images.githubusercontent.com/40359487/113759133-b944c800-96e2-11eb-8647-f36eb9391e68.png)

</div>

**Reply by koegel** [reply]
IMHO the confusion stems from the dialog that pops up for new users which informs them about Gitpod switching to VS Code:
![Selection_358](https://user-images.githubusercontent.com/1044916/115111705-d5b5ef80-9f81-11eb-970e-119142a3759d.png)
I also stumbled upon this and reported it in [this issue](https://github.com/eclipse-theia/theia-website/issues/148) and found it as confusing as @GitMensch. Also I agree with @vince-fugnitto that in the end you get to a Theia instance as expected,  however the way there through the dialog is confusing and I suspect many users might even abort after seeing the dialog since they do not want to try VS Code but Eclipse Theia.
